src/Pages/SignUp/Agreement/Agreement.jsx

```diff
@@ -50,27 +56,28 @@ const Agreement = (props) => {
     }
     if (!comparison.length && usableId) {
       alert('가입을 축하드립니다!');
-      props.history.push('/main');
+      history.push('/main');
     }
     if (!usableId) {
       alert('필명 중복 체크 해주세요.');
     }
   };
```

Comment:
```suggestion
  const goToMain = () => {
    const comparison = checkBox.filter((el) => {
      return el.isCheck
    });
    
    comparison.length
      ? alert(`'${comparison[0].comment}'를 체크해주세요.`);
      : props.history.push('/main');
  };
```
- 변수 선언할 때 가급적 const로 선언해주세요
- filter 안쪽에서 필터 여부 판단하는 조건문 조금 어색해보이는데 el.isCheck가 false일 경우 제거하는게 의도였다면, 다음과 같이 작성해보세요.
- 삼항연산자를 사용해볼 수 있습니다.
- 리뷰 다는 사이에 새로 올라왔군요... 암튼 확인해주세요...